src/transformers/modeling_utils.py

```diff
@@ -2876,10 +2876,18 @@ def to(self, *args, **kwargs):
             raise ValueError("`.to` is not supported for HQQ-quantized models.")
         # Checks if the model has been loaded in 8-bit
         if getattr(self, "quantization_method", None) == QuantizationMethod.BITS_AND_BYTES:
-            raise ValueError(
-                "`.to` is not supported for `4-bit` or `8-bit` bitsandbytes models. Please use the model as it is, since the"
-                " model has already been set to the correct devices and casted to the correct `dtype`."
-            )
+            if self.is_loaded_in_4bit:
```

Comment:
```suggestion
            if getattr(self, "is_loaded_in_4bit", False):
```

When loading in 8bit this attribute isn't defined and will raise an AttributeError.

Reply:
Indeed thanks !